packages/imperative/src/auth/handlers/BaseAuthHandler.ts

```diff
@@ -221,7 +222,8 @@ export abstract class BaseAuthHandler implements ICommandHandler {
 
         const sessCfgWithCreds = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(
             sessCfg, params.arguments,
-            { requestToken: false }
+            { requestToken: false },
+            params
         );
```

Comment:
isn't this a bit strange?
why are we passing `params.arguments` and `params` on the same function?
I know we want to maintain backwards compatibility, but isn't this creating technical debt?

Reply:
id rather change it, not have the technical debt, and break the api - if you're onboarding ill do it 😄

Reply:
I'm not sure how to proceed here.  

This is an SDK method but has a lot of Integration around CLI behavior.  

It will conditionally prompt or call back for input depending on options.  

It's probably a little awkward to use already since it accepts command arguments (which may or may not be applicable).  

Adding an extra full handler object as an optional cleans things up for our internal usage but bloats the API method signature. 

What do you think is the best way to proceed?

Reply:
@zFernand0 - how about changing the method signature to pass handler object as optional in the options object. 

Looking at the code more closely and thinking about this, the options object is currently how we control behavior between other SDK clients and the CLI itself.